src/test/java/uk/ac/cam/cl/dtg/isaac/dos/eventbookings/PgEventBookingsTest.java

```diff
@@ -100,4 +109,61 @@ void getEventBookingStatusCounts_checkEventBookingStatusCounts_canCopeWithCompli
     assertEquals(expectedStatusCounts, actualStatusCounts, "Every row should be represented in the result");
     verify(mockedObjects);
   }
+
+  @Test
+  void testFindAllByEventIds() throws SQLException, SegueDatabaseException {
+    List<String> eventIds = Arrays.asList("event1", "event2");
+    String expectedQuery = "SELECT event_bookings.* FROM event_bookings JOIN users ON users.id=user_id WHERE event_id IN (?, ?) AND NOT users.deleted";
+    Instant now = now();
+
+    EventBooking expectedBooking1 =
+        new PgEventBooking(1L, 1L, 7L, "event1", BookingStatus.CONFIRMED, now, now, null);
+    EventBooking expectedBooking2 =
+        new PgEventBooking(2L, 2L, 7L, "event2", BookingStatus.CONFIRMED, now, now, null);
+
+    expect(dummyPostgresSqlDb.getDatabaseConnection()).andReturn(dummyConnection);
+    expect(dummyConnection.prepareStatement(expectedQuery)).andReturn(dummyPreparedStatement);
+    dummyPreparedStatement.setString(1, "event1");
+    dummyPreparedStatement.setString(2, "event2");
+    expect(dummyPreparedStatement.executeQuery()).andReturn(dummyResultSet);
+
+    expect(dummyResultSet.next()).andReturn(true);
+    expect(dummyResultSet.getLong("id")).andReturn(1L);
+    expect(dummyResultSet.getLong("user_id")).andReturn(1L);
+    expect(dummyResultSet.getLong("reserved_by")).andReturn(7L);
+    expect(dummyResultSet.getString("event_id")).andReturn("event1");
+    expect(dummyResultSet.getString("status")).andReturn("CONFIRMED");
+    expect(dummyResultSet.getTimestamp("created")).andReturn(Timestamp.from(now));
+    expect(dummyResultSet.getTimestamp("updated")).andReturn(Timestamp.from(now));
+    expect(dummyResultSet.getObject("additional_booking_information")).andReturn(null);
+
+    expect(dummyResultSet.next()).andReturn(true);
+    expect(dummyResultSet.getLong("id")).andReturn(2L);
+    expect(dummyResultSet.getLong("user_id")).andReturn(2L);
+    expect(dummyResultSet.getLong("reserved_by")).andReturn(7L);
+    expect(dummyResultSet.getString("event_id")).andReturn("event2");
+    expect(dummyResultSet.getString("status")).andReturn("CONFIRMED");
+    expect(dummyResultSet.getTimestamp("created")).andReturn(Timestamp.from(now));
+    expect(dummyResultSet.getTimestamp("updated")).andReturn(Timestamp.from(now));
+    expect(dummyResultSet.getObject("additional_booking_information")).andReturn(null);
+
+    expect(dummyResultSet.next()).andReturn(false);
+    dummyResultSet.close();
+    dummyPreparedStatement.close();
+    dummyConnection.close();
+
+    // Create an array of all mocked objects
```

Comment:
You can get rid of this comment, as you have already chosen a descriptive variable name and the code you've written is self-explanatory.